apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1645,7 +1681,18 @@ func (u *UniterAPI) NetworkInfo(args params.NetworkInfoParams) (params.NetworkIn
 		}
 	}
 
-	networkInfos := machine.GetNetworkInfoForSpaces(spaces)
+	// Don't get the network info for the default space if there was an error
+	// getting the address.
+	var defaultSpaceNetworkErr error
+	if spaceAddressErr != nil && spaces.Contains(environs.DefaultSpaceName) {
+		spaces.Remove(environs.DefaultSpaceName)
+		defaultSpaceNetworkErr = errors.Annotatef(err, "getting machine %q preferred primary address", machine.MachineTag())
+	}
+
+	networkInfos := machine.GetNetworkInfoForSpaces(spaces, defaultSpaceAddress)
```

Comment:
it feels wrong to be passing an address into machine.GetNetworkInfoForSpaces. can we just remove the default space name from the `spaces` set, and add the address to the result? I'm not even sure if it makes sense to say that the cross-model relation address is in the default space. might need jam to weigh in here...